gpustack/policies/candidate_selectors/base_candidate_selector.py

```diff
@@ -883,3 +884,38 @@ async def _validate_distributed_limit_per_worker(self, worker: Worker) -> bool:
                 return False
 
         return True
+
+    def _check_tp_size_divisibility(
+        self,
+        tp_size: int,
+    ):
+        """
+        Check whether InferenceBackend's constraint of parameter divisibility is satisfied.
+        1. num_attention_heads
+        2. vocab_size
+
+        Notes on `tp_size` (tensor parallel size) usage in auto scheduling:
+        - Single-worker multi-GPU: `tp_size` is the number of GPUs currently selected
+          in the traversal (i.e., `gpu_sum`). The scheduler grows the candidate set
+          incrementally and validates divisibility at each step.
+        - Multi-worker multi-GPU: selected workers are constrained to have the same
+          number of GPUs, so `tp_size` equals the per-worker GPU count (i.e., `gpu_count`)
+        """
+        if not tp_size:
+            return
+        if self._num_attention_heads and self._num_attention_heads % tp_size != 0:
+            raise ValueError(
+                f"Total number of attention heads ({self._num_attention_heads})"
+                " must be divisible by tensor parallel size "
+                f"({tp_size})."
+            )
+
+        if (
+            self._model_params.vocab_size
+            and self._model_params.vocab_size % tp_size != 0
+        ):
+            raise ValueError(
+                f"Vocabulary size ({self._model_params.vocab_size})"
+                " must be divisible by tensor parallel size "
+                f"({tp_size})."
```

Comment:
If this error occurs during automatic scheduling, what should users be advised to do?